File: BackJoon/dp/11053.py
import sys
sys.setrecursionlimit(10**6)

if __name__ == "__main__":
    size = int(sys.stdin.readline())
    numbers = list(map(int, sys.stdin.readline().split(" ")))

    answer = 0
    dp = [0 for _ in range(size)]
    for i in range(size):
        now = 0
        for j in range(i + 1):
            if numbers[i] > numbers[j]:
                now = max(now, dp[j])
        dp[i] = now + 1
        answer = max(answer, dp[i])
    print(answer)

File: BackJoon/dp/11053.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int series[1000000];
int dp[1000000] = {0};

int main() {

    int seriesSize;
    cin >> seriesSize;
    for (int i = 0; i < seriesSize; ++i) {
        cin >> series[i];
    }

    int answer = 0;
    for (int i = 0; i < seriesSize; ++i) {
        int now = 0;
        for (int j = 0; j < i; ++j) {
            if (series[i] > series[j]) {
                now = max(now, dp[j]);
            }
        }
        dp[i] = now + 1;
        answer = max(answer, dp[i]);
    }

    cout << answer << endl;

    return 0;
}